app/utils/clustering/clustering_mvp.py

```diff
@@ -5,105 +5,50 @@
     remainder = num_submissions % 4
-    clusters = []
-
-    # Edge Cases:
-    # - less than 4, they are all in one group
-    if num_submissions < 4:
-        return "Not enough submissions in this cohort to form a full cluster"
-
-    # If the remainder is 3 -> last group will be a group of 3 users
-    elif remainder == 3:
-        # Cluster submissions until 7 remain
-        for i in range(num_clusters - 1):
-            # Group by top 4 squad scores
-            clusters.append(list(df.index[:4]))
-            # Drop stories you have grouped already
-            df = df[4:]
-
-        # Manually cluster final two groups to handle remainder problem
-        # by duplicating 1 submission so each have 4 submission_ids
-        clusters.append(list(df.index[:4]))
-        clusters.append(list(df.index[3:]))
-
-    # If the remainder is 2 -> last 2 groups will be groups of 3
-    elif remainder == 2:
-        if num_submissions == 6:
-            clusters.append(list(df.index[:4]))
-            clusters.append(list(df.index[2:]))
-        else:
-            # Cluster submissions until 10 remain
-            for i in range(num_clusters - 2):
-                # Group by top 4 squad scores
-                clusters.append(list(df.index[:4]))
-                # Drop stories you have grouped already
-                df = df[4:]
-
-            # Manually cluster final three groups to handle remainder problem
-            clusters.append(list(df.index[:4]))
-            clusters.append(list(df.index[3:7]))
-            clusters.append(list(df.index[6:]))
-
-    # If the remainder is 1 -> last 3 groups will be groups of 3
-    elif remainder == 1:
-        if num_submissions == 5:
-            # Overlap both clusters by 3 submissions
-            clusters.append(list(df.index[:4]))
-            clusters.append(list(df.index[1:]))
-        elif num_submissions == 9:
-            clusters.append(list(df.index[:4]))
-            clusters.append(list(df.index[2:6]))
-            clusters.append(list(df.index[5:]))
-        else:
-            # Cluster submissions until 13 remain
-            for i in range(num_clusters - 3):
-                # Group by top 4 squad scores
-                clusters.append(list(df.index[:4]))
-                # Drop stories you have already grouped
-                df = df[4:]
-
-            # Manually cluster final three overlapping groups to handle
-            # remainder problem
-            clusters.append(list(df.index[:4]))
-            clusters.append(list(df.index[3:7]))
-            clusters.append(list(df.index[6:10]))
-            clusters.append(list(df.index[9:]))
-
-    # Else, the remainder is 0. Split evenly by 4
-    else:
-        for i in range(num_clusters):
-            # Group by top 4 squad scores
-            clusters.append(list(df.index[:4]))
-            # Drop stories you have already grouped
-            df = df[4:]
-
-    return clusters
+    matching_minimum = 8
```

Comment:
this should be a constant that probably lives at a higher scope than here. You could put it on the top level on line 8 and give it a name like `MATCHING_MINIMUM`. This is more pythonic

Reply:
Pythonized (is that a word?) the variable based on scope.